src/views/SignUp/SignUp.js

```diff
@@ -0,0 +1,145 @@
+import React,{Component} from "react";
+import Layout from '../../components/Layout';
+
+export default class SignUp extends Component{
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      Lastname: '',
+      email: '',
+      username:'',
+      password:''
+    };
+
+    this.handleSubmitFields = this.handleSubmitFields.bind(this);
+    this.VerifyString = this.VerifyString.bind(this);
+    this.VerifyEmail = this.VerifyEmail.bind(this);
+    this.verifyPassword = this.verifyPassword.bind(this);
+    this.comparePassword = this.comparePassword.bind(this);
+  }
+
+  handleSubmitFields(event) {
+
+    if(this.state.includes(null)){
+      return;
+    }
+
+    var url = 'https://localhost:5001/api/workers/';
+
+    fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(this.state),
+      headers:{
+        'Content-Type': 'application/json'
+      }
+    }).then(res => res.json())
+    .then(response => console.log('Success:', JSON.stringify(response)))
+    .catch(error => console.error('Error:', error));
+    event.preventDefault();
+  }
+
+  VerifyString({ target }) {
+    const {name} = target;
+    if(name.length<=0){
+      console.log('is empty')
+      return;
+    }
+    console.log(name+' exsists')
+
+    this.setState({
+      [name]: target.value
+    });
+  }
+
+  VerifyEmail({ target }) {
```

Comment:
Methods in js shall be camelCased.

Reply:
This is not a valid email validation, we need to use a regex to ensure all emails are accepted.

Reply:
The regex below should validate emails. Add this to a variable and run the input against the regex.
```js
/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)\|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])\|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```